.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,56 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Checkstyle, Findbugs, Doc Check, etc.
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  pull_request:
+    branches: [ github_actions, master, branch-2.* ]
```

Comment:
i was a bit lax in my previous comment. we need to run PR builds for the release branches but only starting from 2.4. i presume none of this would work for 2.3 or earlier, unless you think the full set of changes to make this work can be backported

Reply:
Since the yml files don't exist in the branches themselves, it won't actually run them. For example, note that this code is in the github_actions branch in alluxio/alluxio but it's not actually running in master branch for others right now!

Reply:
good point. so why is this line here at all?

Reply:
ping

Reply:
because the line is still required for it to run. Even with this code being checked in, without this line of branch-2.* it will not run on it.